sycl/include/sycl/detail/info_desc_helpers.hpp

```diff
@@ -95,6 +95,9 @@ struct IsSubGroupInfo<info::kernel_device_specific::compile_sub_group_size>
     using return_type = info::DescType::Desc::return_type;                     \
   };
 #include <sycl/info/device_traits.def>
+// requires explicit handling
+__SYCL_PARAM_TRAITS_SPEC(device, parent_device, sycl::device,
+                         PI_DEVICE_INFO_PARENT_DEVICE)
```

Comment:
A better approach/pattern to such special handling is, IMO:

```
// .def:
#ifndef __SYCL_PARAM_TRAITS_SPEC_SPECIAL(...)
#define __SYCL_PARAM_TRAITS_SPEC_SPECIAL(...) __SYCL_PARAM_TRAITS_SPEC(...)
#endif

// use_normal.cpp
#define __SYCL_PARAM_TRAITS_SPEC(....) ...
#include 

// use_special.cpp
#define __SYCL_PARAM_TRAITS_SPEC(....) ....
#define __SYCL_PARAM_TRAITS_SPEC_SPECIAL(...)
#include
```

That said, I'm not familiar with the code to comment on whether it does the right thing. I can only review how it does what it's doing.

Reply:
That's a good suggestion. I was hurrying just trying to fix the post-commit failure. Let me check it out. If it's easy enough, I'll do it now. If not, I'll open a JIRA and do it next week.

Reply:
done